hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MetricsCompressor.java

```diff
@@ -329,13 +339,32 @@ private void reset(int estimatedBytesDictionary, int estimatedBytesMetrics) {
         lastDescriptor = null;
     }
 
-    private byte[] getRenderedBlob() {
+    /**
+     * Frees resources associated with this compressor. Needed to avoid keeping
+     * memory allocated by {@link Deflater} for a long time, until finalization.
+     */
+    public void close() {
         try {
-            writeDictionary();
             dictionaryDos.close();
-            dictionaryCompressor.end();
+            if (dictionaryCompressor != null) {
+                dictionaryCompressor.end();
+                dictionaryCompressor = null;
+            }
             metricDos.close();
-            metricsCompressor.end();
+            if (metricsCompressor != null) {
+                metricsCompressor.end();
+                metricsCompressor = null;
+            }
```

Comment:
I don't understand why we need to nullify `Deflater` references. If you are worried that the native memory referenced by `Deflater#zsRef` might not be released, [Deflater.end() is supposed to release it][1].

[1]: https://bugs.openjdk.org/browse/JDK-8257032

Reply:
I was lazy and did not want to check if you can run `Deflater.end()` multiple times